### PR TITLE
Add phpstan and psalm

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is BC.

## Changelog

```markdown
### Added
- Added PHPStan and Psalm
```

### CODE REVIEW

File: phpstan-baseline.neon [hunk 1]
```diff
@@ -1,6 +1,16 @@
 parameters:
-	ignoreErrors:
-		-
-			message: "#^Class Sonata\\\\UserBundle\\\\Model\\\\User not found\\.$#"
-			count: 2
-			path: src/Timezone/UserBasedTimezoneDetector.php
+    ignoreErrors:
+        -
+            message: "#^Class Sonata\\\\UserBundle\\\\Model\\\\User not found\\.$#"
```
Comment (VincentLanglet) [inline phpstan-baseline.neon:9]: Can you add a comment to explain why userBundle is not a dev-dependency ? :)

Comment (jordisala1991) [reply phpstan-baseline.neon:9]: Added

**Review by VincentLanglet — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Isn't it pedantic ?

**Review by core23 — APPROVED** [review]

**Comment by core23** [comment]
> Isn't it pedantic ?

IMHO it's a patch, because there are some tiny code changes.

**Comment by jordisala1991** [comment]
Needs dev kit Pr

**Pull request merged**